crash when using an unknown resource name with `limesctl project set`

To reproduce, run `limes project set $SOME_PROJECT_ID object-storage/capacity=$SOME_VALUE`. Note that `object-store` is misspelled as `object-storage`. This segfaults for me with the following trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x7a50b5]

goroutine 1 [running]:
github.com/sapcc/limesctl/pkg/cli.(*Project).setBaseUnits(0xc0000c2000, 0xc0003f8090, 0xc000121d00)
        /x/src/github.com/sapcc/limesctl/.gopath/src/github.com/sapcc/limesctl/pkg/cli/parse_quotas.go:265 +0x205
github.com/sapcc/limesctl/pkg/cli.ParseRawQuotas(0x923a20, 0xc0000c2000, 0xc00000e9e0, 0x0, 0x0, 0x0, 0x0)
        /x/src/github.com/sapcc/limesctl/.gopath/src/github.com/sapcc/limesctl/pkg/cli/parse_quotas.go:156 +0x5ad
main.main()
        /x/src/github.com/sapcc/limesctl/.gopath/src/github.com/sapcc/limesctl/main.go:253 +0x6ce
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,7 @@ Bugfixes:
 - On Windows, handle UTF-8-encoded environment variables in the same way as
   python-openstackclient. See
   <https://github.com/gophercloud/gophercloud/issues/1572> for details.
+- Do not crash when unknown service/resource are used with `set` subcommand.
 
 # v1.5.3 (2019-08-20)
 

--- a/internal/auth/client.go
+++ b/internal/auth/client.go
@@ -5,8 +5,8 @@ import (
 
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack"
-	"github.com/gophercloud/utils/openstack/clientconfig"
 	"github.com/gophercloud/utils/client"
+	"github.com/gophercloud/utils/openstack/clientconfig"
 	"github.com/sapcc/gophercloud-limes/resources"
 	"github.com/sapcc/limesctl/internal/errors"
 )
@@ -15,28 +15,22 @@ import (
 // for Keystone and Limes.
 func ServiceClients(debug bool) (identityV3, limesV1 *gophercloud.ServiceClient) {
 	ao, err := clientconfig.AuthOptions(nil)
-	if err != nil {
-		errors.Handle(err, "can not get auth variables")
-	}
+	errors.Handle(err, "can not get auth variables")
 
 	provider, err := openstack.NewClient(ao.IdentityEndpoint)
-	if err != nil {
-		errors.Handle(err, "can not create an OpenStack client")
-	}
+	errors.Handle(err, "can not create an OpenStack client")
 
 	if debug {
 		provider.HTTPClient = http.Client{
 			Transport: &client.RoundTripper{
-				Rt: &http.Transport{},
+				Rt:     &http.Transport{},
 				Logger: &client.DefaultLogger{},
 			},
 		}
 	}
 
 	err = openstack.Authenticate(provider, *ao)
-	if err != nil {
-		errors.Handle(err, "can not connect to OpenStack")
-	}
+	errors.Handle(err, "can not connect to OpenStack")
 
 	identityClient, err := openstack.NewIdentityV3(provider, gophercloud.EndpointOpts{})
 	errors.Handle(err, "could not initialize identity client")

--- a/internal/core/parse_quotas.go
+++ b/internal/core/parse_quotas.go
@@ -192,16 +192,18 @@ func (c *Cluster) setBaseUnits(limesV1 *gophercloud.ServiceClient, ru *resourceU
 	cluster, err := c.Result.Extract()
 	errors.Handle(err, "cluster does not exist")
 
-	for srv, resMap := range *ru {
-		for res := range resMap {
-			if cluster.Services[srv] != nil {
-				srvRes, exists := cluster.Services[srv].Resources[res]
-				if exists {
-					(*ru)[srv][res] = srvRes.ResourceInfo.Unit
-				}
-			} else {
-				errors.Handle(fmt.Errorf("service %s for cluster %s could not be found", srv, cluster.ID), "")
+	for srvName, resMap := range *ru {
+		for resName := range resMap {
+			srvReport, exists := cluster.Services[srvName]
+			if !exists {
+				errors.Handle(fmt.Errorf("service %q for cluster %q could not be found", srvName, cluster.ID), "")
 			}
+			resReport, exists := srvReport.Resources[resName]
+			if !exists {
+				errors.Handle(fmt.Errorf("resource %q for service %q for cluster %q could not be found", resName, srvName, cluster.ID), "")
+			}
+
+			(*ru)[srvName][resName] = resReport.ResourceInfo.Unit
 		}
 	}
 }
@@ -220,16 +222,18 @@ func (d *Domain) setBaseUnits(limesV1 *gophercloud.ServiceClient, ru *resourceUn
 	domain, err := d.Result.Extract()
 	errors.Handle(err, "domain does not exist")
 
-	for srv, resMap := range *ru {
-		for res := range resMap {
-			if domain.Services[srv] != nil {
-				srvRes, exists := domain.Services[srv].Resources[res]
-				if exists {
-					(*ru)[srv][res] = srvRes.ResourceInfo.Unit
-				}
-			} else {
-				errors.Handle(fmt.Errorf("service %s for domain %s could not be found", srv, domain.Name), "")
+	for srvName, resMap := range *ru {
+		for resName := range resMap {
+			srvReport, exists := domain.Services[srvName]
+			if !exists {
+				errors.Handle(fmt.Errorf("service %q for domain %q could not be found", srvName, domain.Name), "")
 			}
+			resReport, exists := srvReport.Resources[resName]
+			if !exists {
+				errors.Handle(fmt.Errorf("resource %q for service %q for domain %q could not be found", resName, srvName, domain.Name), "")
+			}
+
+			(*ru)[srvName][resName] = resReport.ResourceInfo.Unit
 		}
 	}
 }
@@ -248,16 +252,18 @@ func (p *Project) setBaseUnits(limesV1 *gophercloud.ServiceClient, ru *resourceU
 	project, err := p.Result.Extract()
 	errors.Handle(err, "project does not exist")
 
-	for srv, resMap := range *ru {
-		for res := range resMap {
-			if project.Services[srv] != nil {
-				srvRes, exists := project.Services[srv].Resources[res]
-				if exists {
-					(*ru)[srv][res] = srvRes.ResourceInfo.Unit
-				}
-			} else {
-				errors.Handle(fmt.Errorf("service %s for project %s could not be found", srv, project.Name), "")
+	for srvName, resMap := range *ru {
+		for resName := range resMap {
+			srvReport, exists := project.Services[srvName]
+			if !exists {
+				errors.Handle(fmt.Errorf("service %q for project %q could not be found", srvName, project.Name), "")
 			}
+			resReport, exists := srvReport.Resources[resName]
+			if !exists {
+				errors.Handle(fmt.Errorf("resource %q for service %q for project %q could not be found", resName, srvName, project.Name), "")
+			}
+
+			(*ru)[srvName][resName] = resReport.ResourceInfo.Unit
 		}
 	}
 }

